ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+                    import matplotlib as mpl
+
+                    norm_color = mpl.colors.Normalize(
+                        vmin=limits["z_min"], vmax=limits["z_max"]
+                    )
+                    cb = ax[0][-1].figure.colorbar(
+                        mpl.cm.ScalarMappable(
+                            norm=norm_color, cmap=plot_options["colormap"]),
+                        ax=ax[0][-1],
+                        location="top",
+                        fraction=0.1,
+                        label="Score",
+                    )
+                    cb.ax.locator_params(nbins=8)
+
+                    # Add a legend for the various figure contents
+                    if isinstance(pars, str):
+                        if pars == "result":
+                            highlight_label = "Best data point"
+                        elif pars == "expected_minimum":
+                            highlight_label = "Expected minimum"
+                        elif pars == "expected_minimum_random":
+                            highlight_label = "Simulated minimum"
+                    elif isinstance(pars, list):
+                        # The case where the user specifies [x[0], x[1], ...]
+                        highlight_label = "Point: " + str(pars)
+                    # Legend icon for data points
+                    legend_data_point = mpl.lines.Line2D(
+                        [],
+                        [],
+                        color="darkorange",
+                        marker=".",
+                        markersize=9,
+                        lw=0.0,
+                    )
+                    # Legend icon for the highlighted point in the 2D plots
+                    legend_hp = mpl.lines.Line2D(
+                        [],
+                        [],
+                        color="k",
+                        marker="D",
+                        markersize=5,
+                        lw=0.0,
+                    )
+                    # Legend icon for the highlighted value in the 1D plots
+                    legend_hl = mpl.lines.Line2D(
+                        [],
+                        [],
+                        linestyle="--",
+                        color="k",
+                        marker="",
+                        lw=1,
+                    )
+                    if show_confidence:
+                        # Legend icon for the 95 % credibility interval
+                        legend_fill = mpl.patches.Patch(
+                            color="green",
+                            alpha=0.5,
+                        )
+                        if usepartialdependence:
+                            ci_label = "Est. 95 % credibility interval"
+                        else:
+                            ci_label = "95 % credibility interval"
+                        ax[0][-1].legend(
+                            handles=[legend_data_point,
```

Comment:
Do we really need the strange newline construction here? If you want the contents of the `handles` list to appear legible we should at least split them across a new line for each entry, instead of just having a single new line after the first argument. The way this is now structured makes it harder to decode the individual arguments for the legend call.